omod/src/main/java/org/openmrs/module/reporting/web/datasets/ManageDatasetsController.java

```diff
@@ -27,37 +31,64 @@
-    		ModelMap model) {
 	
-		DataSetDefinitionService service = Context.getService(DataSetDefinitionService.class);
-		DataSetDefinition dsd = service.getDefinition(uuid, type);
-	 	model.addAttribute("dataSetDefinition", dsd);
-	
-	 	List<Property> properties = DefinitionUtil.getConfigurationProperties(dsd);
-	 	model.addAttribute("configurationProperties", properties);
-	 	Map<String, List<Property>> groups = new LinkedHashMap<String, List<Property>>();
-	 	for (Property p : properties) {
-	 		List<Property> l = groups.get(p.getGroup());
-	 		if (l == null) {
-	 			l = new ArrayList<Property>();
-	 			groups.put(p.getGroup(), l);
-	 		}
-	 		l.add(p);
-	 	}
-	 	model.addAttribute("groupedProperties", groups);
+	@SuppressWarnings("unchecked")
+	@RequestMapping("/module/reporting/datasets/editDataSet")
+	 public String editDataSet(
+	    		@RequestParam(required=false, value="uuid") String uuid,
+	            @RequestParam(required=false, value="type") Class<? extends DataSetDefinition> type,
+	            @RequestParam(required=false, value="cohortSize") Integer cohortSize,
+	            @RequestParam(required=false, value="action") String action,
+	    		ModelMap model) {
+		
+			DataSetDefinitionService service = Context.getService(DataSetDefinitionService.class);
+			DataSetDefinition dsd = service.getDefinition(uuid, type);
+			
+		 	model.addAttribute("dataSetDefinition", dsd);
+		
+		 	List<Property> properties = DefinitionUtil.getConfigurationProperties(dsd);
+		 	model.addAttribute("configurationProperties", properties);
+		 	Map<String, List<Property>> groups = new LinkedHashMap<String, List<Property>>();
+		 	for (Property p : properties) {
+		 		List<Property> l = groups.get(p.getGroup());
+		 		if (l == null) {
+		 			l = new ArrayList<Property>();
+		 			groups.put(p.getGroup(), l);
+		 		}
+		 		l.add(p);
+		 	}
+		 	model.addAttribute("groupedProperties", groups);
+		
+		    return "/module/reporting/datasets/datasetEditor";
+	    }
 	
-	    return "/module/reporting/datasets/datasetEditor";
-    }
+	/*
+	 * Loads Affected Report Definitions.
+	 */
+	@RequestMapping("/module/reporting/datasets/loadAffectedDatasetDefs")
+	@ResponseBody()
+	public List<ReportDefinition> loadAffectedDatasetDefs(@RequestParam(required=true,value="uuid")String uuid){
+		
+		List<ReportDefinition> reportDefinitions = Context.getService(ReportDefinitionService.class).getAllDefinitions(false);	
+		List<ReportDefinition> affectedReportDefs = new ArrayList<ReportDefinition>();
+		for (ReportDefinition reportDef : reportDefinitions) {
+			List<Mapped<? extends DataSetDefinition>> dsds = new ArrayList<Mapped<? extends DataSetDefinition>>(reportDef.getDataSetDefinitions().values());
+			if (!dsds.isEmpty()) {
+				for (Mapped<? extends DataSetDefinition> dataSet : dsds) {
+				     
+					  if (uuid.equals(dataSet.getUuidOfMappedOpenmrsObject())) {
+						  affectedReportDefs.add(reportDef);
+					  }
+				}
+			} 
+		}
+		
+		return affectedReportDefs;
```

Comment:
Before I squash my commits, in this first commit, this controller function returns a list of report defs that depend on this dataset, but Its has bidirectional dependencies coz jackson fails to serialize the ReportDefinition object with this [error](https://dpaste.de/kDKV) and I this this is coz of mappings in this class [BaseOpenmrsMetadata.java](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/BaseOpenmrsMetadata.java#L44) yet the [User class](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/User.java) extends the BaseOpenmrsMetadata.java class. This causes the error. And I think that needs some work in the OpenMRS core API to solve this. Thats why I came up with an inner private class in the  controller class.
Please will this work? cc. @mseaton ,@dkayiwa

Reply:
Seems I haven't understood what you mean by "if the creator of a user is the same user. Where did this test data come from?" otherwise everything is fine now @mseaton